test/observe.lean

```diff
@@ -4,14 +4,19 @@ import Mathlib.Data.Nat.Prime
 
 open Nat
 
+set_option maxHeartbeats 150000 in
 theorem euclid (n : ℕ) : ∃ N, n < N ∧ N.Prime := by
   let N := n.factorial + 1
   let p := minFac N
   use p
   have prime : p.Prime := by
-    apply minFac_prime
+    -- we write this in a funny way,
+    -- so that we can test that `observe` (aka `apply?`)
+    -- finds `minFac_prime` in time
     observe : n.factorial > 0
-    linarith
+    have : N ≠ 1 := by linarith
+    observe : p.Prime
+    assumption
```

Comment:
I don't understand this change, what is going on?

Reply:
I want a way to test that `apply?` finds `minFac_prime` (fast). This is my best approximation of testing that. Ideally we should test that it appears amongst the first 10 suggestions.

Reply:
Could you explain *how* it tests that? I still find this completely inscrutable. :-)

Reply:
Can you just wrap `observe` or `apply?` with a `show_term`? You could even wrap that with a `#guard_msgs` (maybe we have a tactic version?).

Reply:
Aha, I guess we would need a tactic version of `guard_msgs`... but I wouldn't know how to make this precise.

Reply:
I *still* do not understand why you care about this. I would much rather this file contains a nice easy proof than an inscrutable test.

Reply:
I've reverted those changes.